libraries/chain/database_worker_objects.cpp

```diff
@@ -156,34 +156,56 @@ namespace golos { namespace chain {
         const auto& wto_idx = get_index<worker_techspec_index, by_next_cashout_time>();
 
         for (auto wto_itr = wto_idx.begin(); wto_itr != wto_idx.end() && wto_itr->next_cashout_time <= now; ++wto_itr) {
-            const auto& wpo = get_worker_proposal(wto_itr->worker_proposal_author, wto_itr->worker_proposal_permlink);
+            auto remaining_payments_count = wto_itr->payments_count - wto_itr->finished_payments_count;
 
-            const auto& reward = wto_itr->development_cost / wto_itr->payments_count;
+            auto author_reward = wto_itr->specification_cost / wto_itr->payments_count;
+            auto author_remaining = wto_itr->specification_cost - (author_reward * wto_itr->finished_payments_count);
+            author_reward = author_remaining / remaining_payments_count;
+
+            auto worker_reward = wto_itr->development_cost / wto_itr->payments_count;
+            auto worker_remaining = wto_itr->development_cost - (worker_reward * wto_itr->finished_payments_count);
+            worker_reward = worker_remaining / remaining_payments_count;
 
             const auto& gpo = get_dynamic_global_properties();
-            modify(gpo, [&](dynamic_global_property_object& gpo) {
-                gpo.total_worker_fund_steem -= reward;
-            });
 
-            adjust_balance(get_account(wto_itr->worker), reward);
+            if (gpo.total_worker_fund_steem < (author_reward + worker_reward)) {
+                return;
+            }
 
             if (wto_itr->finished_payments_count+1 == wto_itr->payments_count) {
                 modify(*wto_itr, [&](worker_techspec_object& wto) {
                     wto.finished_payments_count++;
                     wto.next_cashout_time = time_point_sec::maximum();
                 });
 
+                const auto& wpo = get_worker_proposal(wto_itr->worker_proposal_author, wto_itr->worker_proposal_permlink);
                 modify(wpo, [&](worker_proposal_object& wpo) {
                     wpo.state = worker_proposal_state::closed;
                 });
+
+                auto consumption = (wto_itr->development_cost + wto_itr->specification_cost) * fc::days(30).to_seconds()
+                    / (uint64_t(wto_itr->payments_interval) * wto_itr->payments_count);
+                consumption = std::min(consumption, wto_itr->development_cost + wto_itr->specification_cost);
```

Comment:
From what cases protect this expression?

Reply:
@afalaleev 
payments_interval = 3600 sec
payments_count = 2
cost = 7200 GOLOS

But without protection we have:
1) per second = 1 GOLOS
2) per month = 2592000 GOLOS

With protection we have:
3) std::min(2592000, 7200) = 7200